src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -22,6 +23,90 @@ namespace System
     /// </summary>
     public static class Span
     {
+        /// <summary>
+        /// Determines whether the beginning of the span matches the specified value when compared using the specified comparison option.
+        /// </summary>
+        public static bool StartsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison)
+        {
+            if ((uint)(comparison - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
+            {
+                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparison));
+            }
+
+            if (span.Length < value.Length)
+            {
+                return false;
+            }
+
+            switch (comparison)
+            {
+                case StringComparison.CurrentCulture:
+                    return CultureInfo.CurrentCulture.CompareInfo.IsPrefix(span, value, CompareOptions.None);
+
+                case StringComparison.CurrentCultureIgnoreCase:
+                    return CultureInfo.CurrentCulture.CompareInfo.IsPrefix(span, value, CompareOptions.IgnoreCase);
+
+                case StringComparison.InvariantCulture:
+                    return CompareInfo.Invariant.IsPrefix(span, value, CompareOptions.None);
+
+                case StringComparison.InvariantCultureIgnoreCase:
+                    return CompareInfo.Invariant.IsPrefix(span, value, CompareOptions.IgnoreCase);
+
+                case StringComparison.Ordinal:
+                    // TODO: https://github.com/dotnet/corefx/issues/25182
+                    // return span.StartsWith(value);
+                    throw new NotImplementedException();
```

Comment:
Rather than throwing, can you put a basic implementation in place now and then follow-up subsequently with whatever changes/improvements you want?  It should only be a few lines.

Reply:
right the implementation should be straight forward

---
In reply to: [166248055](https://github.com/dotnet/coreclr/pull/16223#discussion_r166248055) [](ancestors = 166248055)